<launch>


  <arg name="only_use_tracker" value="false"/>
  <arg name="partial_mosaic_dur" value="3.5"/>
  <node pkg="rosbag" type="play" name="player" args="--pause -r 1 -s 0 -u $(arg partial_mosaic_dur) $(find dvs_mosaic)/data/synth2/events.bag" />
  <node name="mosaic" pkg="dvs_mosaic" type="dvs_mosaic" output="screen" args="--v 1">
    
    <param name="new_dataset_"      value="true"/>

    <param name="display_accuracy_"      value="true"/>
    <param name="tracker_standalone_"    value="$(arg only_use_tracker)"/>
    <param name="use_partial_mosaic_"    value="true"/>
    <param name="partial_mosaic_dur_"    value="$(arg partial_mosaic_dur)"/>

    <param name="mosaic_height_"      value="512"/>
    <param name="init_packet_num_"    value="20000"/>
   
    <param name="num_events_update_"            value="500"/>
    <param name="num_packet_reconstrct_mosaic_" value="25"/>

    <!-- Mapper parameters -->
    <param name="num_events_map_update_"  value="500"/>
    <param name="measure_contrast_"       value="true"/>
    <param name="var_R_mapping_"          value="0.0289"/>
    <param name="use_gaussian_blur_"      value="false"/>
    <param name="gaussian_blur_sigma_"    value="0.5"/>
    <!-- Pose average -->
    <param name="average_pose_"           value="false"/>
    <param name="average_method_"         value="true"/>
    <param name="average_level_"          value="5"/>

    <!-- Tracker parameters -->
    <param name="num_events_pose_update_" value="500"/>
    <param name="var_process_noise_"      value="1.0e-4"/>
    <param name="var_R_tracking_"         value="1.0e-2"/>
    <!-- Three Thresholds -->
    <param name="use_grad_thres_"         value="true"/>
    <param name="grad_thres_"             value="1.0e-2"/>
    <param name="use_polygon_thres_"      value="true"/>
    <param name="tracking_area_percent_"  value="0.85"/>
    <param name="use_bright_thres_"       value="true"/>
    <param name="bright_thres_"           value="1.0e-2"/>

    <!-- Which topic to listen? Map topic name of input events -->
    <remap from="events" to="/cam0/events" />
    <remap from="pose" to="/cam0/pose" />

  </node>


  <node if="$(arg only_use_tracker)"  name = "image_display" pkg = "rqt_gui" type = "rqt_gui" respawn = "false" output = "screen"
    args = "--perspective-file $(find dvs_mosaic)/launch/mosaic_view_tracker.perspective"/>
  <node unless="$(arg only_use_tracker)"  name = "image_display" pkg = "rqt_gui" type = "rqt_gui" respawn = "false" output = "screen"
    args = "--perspective-file $(find dvs_mosaic)/launch/mosaic_view.perspective"/>

</launch>
